### PR TITLE
adding iptables fixture and controller

### PR DESCRIPTION
**Pull request by nshidlin**
This PR adds an iptables controller and fixture in base_test.

This allows to add and remove iptables rules from within the tests based on node ips

### CODE REVIEW

File: discovery-infra/test_infra/controllers/iptables.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        rule += f' -s {sources_string}' if self._sources else ''
+        rule += f' --dport {self._dest_port}' if self._dest_port else ''
+        rule += f' {self._extra_agrs}' if self._extra_agrs else ''
+        rule += f' -j {self._target}'
+
+        return rule
+    
+    def _does_rule_exist(self):
+        check_rule = self._build_command_string('check')
+        res = run_command(check_rule, shell=True, raise_errors=False)
```
Comment (lalon4) [inline discovery-infra/test_infra/controllers/iptables.py:28]: Assuming the rule doesn't exist, than you apply it. no?

Comment (nshidlin) [reply discovery-infra/test_infra/controllers/iptables.py:28]: this function just checks if the rule exists

This is checked only on delete

You're right will check on insert as well

File: discovery-infra/test_infra/controllers/iptables.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        self._chain = chain
+        self._target = target
+        self._protocol = protocol
+        self._dest_port = dest_port
+        self._sources = sources if sources else []
+        self._extra_agrs = extra_args
+
+    def _build_command_string(self, option):
+        sources_string = ','.join(self._sources)
+        rule = f'iptables --{option}'
```
Comment (YuviGold) [inline discovery-infra/test_infra/controllers/iptables.py:16]: This isn't so readable..  Maybe a core list with `" ".join(...)` and only add the relevant flags if needed? 
i.e.
```python
rule = ["iptables", f"--{option}", self._chain, "-p" ,self._protocol, "-j", self._target]

if self.sources:
    rule += ["-s", sources_string]

if self._dest_port:
    rule += ["--dport", self._dest_port]

if self._extra_args:
    rule += [self._extra_args]

return " ".join(rule_template)

File: discovery-infra/test_infra/controllers/iptables.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+from test_infra.utils import run_command
+
+class IptableRule:
+    def __init__(self, chain, target, protocol, dest_port='', sources=None, extra_args=''):
+        self._chain = chain
+        self._target = target
+        self._protocol = protocol
+        self._dest_port = dest_port
+        self._sources = sources if sources else []
+        self._extra_agrs = extra_args
```
Comment (YuviGold) [inline discovery-infra/test_infra/controllers/iptables.py:12]: typo
`self._extra_agrs` -> `self._extra_args`

File: discovery-infra/test_infra/controllers/iptables.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        rule += f' -s {sources_string}' if self._sources else ''
+        rule += f' --dport {self._dest_port}' if self._dest_port else ''
+        rule += f' {self._extra_agrs}' if self._extra_agrs else ''
+        rule += f' -j {self._target}'
+
+        return rule
+    
+    def _does_rule_exist(self):
+        check_rule = self._build_command_string('check')
+        res = run_command(check_rule, shell=True, raise_errors=False)
```
Comment (YuviGold) [inline discovery-infra/test_infra/controllers/iptables.py:28]:
```python
_, _, exit_code = ....

return exit_code == 0
```

File: discovery-infra/test_infra/controllers/iptables.py (hunk 2 above)
Comment (nshidlin) [reply discovery-infra/test_infra/controllers/iptables.py]: done

File: discovery-infra/test_infra/controllers/iptables.py (hunk 4 above)
Comment (nshidlin) [reply discovery-infra/test_infra/controllers/iptables.py]: done

File: discovery-infra/test_infra/controllers/iptables.py (hunk 3 above)
Comment (nshidlin) [reply discovery-infra/test_infra/controllers/iptables.py]: done

**Review by YuviGold — APPROVED** [review]
LGTM

**Pull request merged**